server/routes/appointments/search/handlers/search.ts

```diff
@@ -18,32 +19,45 @@ export class Search {
 }
 
 export default class SearchRoutes {
-  constructor(private readonly activitiesService: ActivitiesService) {}
+  constructor(private readonly activitiesService: ActivitiesService, private readonly prisonService: PrisonService) {}
 
   GET = async (req: Request, res: Response): Promise<void> => {
     const { user } = res.locals
-    const { startDate, timeSlot, categoryCode, locationId, prisonerNumber, createdBy, type } = req.query
-
-    if (!isValid(toDate(startDate as string))) {
-      return res.redirect(`?startDate=${toDateString(new Date())}`)
-    }
+    const { startDate, timeSlot, categoryCode, locationId, prisonerNumber, createdBy } = req.query
 
     const simpleStartDate = simpleDateFromDate(toDate(startDate as string))
 
-    const categories = await this.activitiesService.getAppointmentCategories(user)
-    const locations = await this.activitiesService.getAppointmentLocations(user.activeCaseLoadId, user)
+    if (!simpleStartDate) {
+      return res.redirect(`?startDate=${toDateString(new Date())}`)
+    }
 
     const request = {
-      appointmentType: type && type !== '' ? type : null,
       startDate: simpleStartDate.toIsoString(),
-      timeSlot: timeSlot && timeSlot !== '' ? timeSlot : null,
-      categoryCode: categoryCode && categoryCode !== '' ? categoryCode : null,
-      internalLocationId: locationId && locationId !== '' ? +locationId : null,
-      prisonerNumbers: prisonerNumber && prisonerNumber !== '' ? [prisonerNumber] : null,
-      createdBy: createdBy && createdBy !== '' && createdBy !== 'all' ? createdBy : null,
+      timeSlot: timeSlot || null,
+      categoryCode: categoryCode || null,
+      internalLocationId: locationId ? +locationId : null,
+      prisonerNumbers: prisonerNumber ? [prisonerNumber] : null,
+      createdBy: createdBy && createdBy !== 'all' ? createdBy : null,
```

Comment:
I've updated the logic here since if `timeSlot` is truthy then it cannot be `''` making the empty string check redundant